### ISSUE TITLE
Problem building against gcc:10/clang:11

### ISSUE THREAD

**Issue by bartoszek** [jarro2783/cxxopts#276]
After #273 was merged [cxxopts-git](https://aur.archlinux.org/packages/cxxopts-git) failst to build on Arch Linux with both gcc and clang 
clang:11.0.1 complains about `unknown warning group '-Werror=null-dereference'`
```
[ 50% | 4<2<3 | 2.3bps] Building CXX object test/CMakeFiles/link_test.dir/link_b.cpp.o
FAILED: test/CMakeFiles/link_test.dir/link_b.cpp.o
/usr/bin/clang++  -I/build/cxxopts-git/src/cxxopts-git/include -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wall -Werror -Wextra -Wshadow -Weffc++ -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self -Wconversion -Wno-sign-conversion -Wsuggest-override -std=c++11 -MD -MT test/CMakeFiles/link_test.dir/link_b.cpp.o -MF test/CMakeFiles/link_test.dir/link_b.cpp.o.d -o test/CMakeFiles/link_test.dir/link_b.cpp.o -c /build/cxxopts-git/src/cxxopts-git/test/link_b.cpp
In file included from /build/cxxopts-git/src/cxxopts-git/test/link_b.cpp:1:
/build/cxxopts-git/src/cxxopts-git/include/cxxopts.hpp:1138:32: error: unknown warning group '-Werror=null-dereference', ignored [-Werror,-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Werror=null-dereference"
                               ^
1 error generated.
```

**Comment by bartoszek** [comment]
gcc:10.2.0 complains about `-Werror=null-dereference’ is not an option that controls warnings`
```
[ 62%] Building CXX object test/CMakeFiles/options_test.dir/options.cpp.o
In file included from /build/cxxopts-git/src/cxxopts-git/test/link_b.cpp:1:
/build/cxxopts-git/src/cxxopts-git/include/cxxopts.hpp:1138:32: error: ‘-Werror=null-dereference’ is not an option that controls warnings [-Werror=pragmas]
 1138 | #pragma GCC diagnostic ignored "-Werror=null-dereference"
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [test/CMakeFiles/link_test.dir/build.make:95: test/CMakeFiles/link_test.dir/link_b.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....

**Comment by bartoszek** [comment]
I quickly patched this for gcc:10 build for AUR deployment, but think it should be probably reported upstream.
https://github.com/bartoszek/AUR-cxxopts-git/commit/a1e7c291ecb57ee74ecb5262f308118419d4fdb0

**Comment by bartoszek** [comment]
As stated by gcc devs this type of `#pragma` is invalid and won't be supported `WONTFIX` in the later gcc version.
So there's _your shtick_ to globally disable `null-dereference` warning=error status not only some parto of the code.
